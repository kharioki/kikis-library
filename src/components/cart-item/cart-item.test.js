import React from "react";
import { shallow } from "enzyme";

import CartItem from "./cart-item.component";

describe("CartItem tests", () => {
  let wrapper;
  let mockClearItem;
  let mockAddItem;
  let mockRemoveItem;

  beforeEach(() => {
    mockClearItem = jest.fn();
    mockAddItem = jest.fn();
    mockRemoveItem = jest.fn();

    const mockProps = {
      cartItem: {
        imageUrl: "www.someimage.com",
        days: 2,
        name: "item test",
        quantity: 1
      },
      clearItem: mockClearItem,
      addItem: mockAddItem,
      removeItem: mockRemoveItem
    };

    wrapper = shallow(<CartItem {...mockProps} />);
  });

  it("should render CartItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
