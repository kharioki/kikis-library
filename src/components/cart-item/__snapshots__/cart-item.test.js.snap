// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem tests should render CartItem component 1`] = `ShallowWrapper {}`;
